textdb/textdb-exp/src/main/java/edu/uci/ics/textdb/exp/nlp/splitter/NlpSplitOperator.java

```diff
@@ -0,0 +1,181 @@
+import edu.uci.ics.textdb.api.tuple.Tuple;
+import edu.uci.ics.textdb.api.utils.Utils;
+
+/**
+ * This Operator splits the input string using Stanford core NLP's ssplit annotation.
+ * 
+ * The result is an list of sentences.
+ * 
+ * The result will be put into an attribute with resultAttributeName specified in predicate, and type List<String>.
+ * 
+ * @author Venkata Raj Kiran Kollimarla, Vinay Bagade
+ *
+ */
+public class NlpSplitOperator implements IOperator {
+    
+    private final NlpSplitPredicate predicate;
+    private IOperator inputOperator;
+    private Schema outputSchema;
+    private int cursor = CLOSED;
+    //A flag to keep track of any remaining sentences from the previous input tuple
+//    private boolean sentenceListIsEmpty = true;
+    //A tuple that persists between method calls
+    private Tuple currentTuple;
+    //A list of sentences generated from the current tuple
+    private List<TextField> currentSentenceList = new ArrayList<TextField>();
+    
+    
+    public NlpSplitOperator(NlpSplitPredicate predicate) {
+        this.predicate = predicate;
+    }
+    
+    public void setInputOperator(IOperator operator) {
+        if (cursor != CLOSED) {  
+            throw new RuntimeException("Cannot link this operator to other operator after the operator is opened");
+        }
+        this.inputOperator = operator;
+    }
+    
+    /*
+     * adds a new field to the schema, with name resultAttributeName and type list of strings
+     */
+    private Schema transformSchema(Schema inputSchema) {
+        if (inputSchema.containsField(predicate.getResultAttributeName())) {
+            throw new RuntimeException(String.format(
+                    "result attribute name %s is already in the original schema %s", 
+                    predicate.getResultAttributeName(),
+                    inputSchema.getAttributeNames()));
+        }
+        return Utils.addAttributeToSchema(inputSchema, 
+                new Attribute(predicate.getResultAttributeName(), AttributeType.LIST));
+    }
+      
+
+    @Override
+    public void open() throws TextDBException {
+        if (cursor != CLOSED) {
+            return;
+        }
+        if (inputOperator == null) {
+            throw new DataFlowException(ErrorMessages.INPUT_OPERATOR_NOT_SPECIFIED);
+        }
+        inputOperator.open();
+        Schema inputSchema = inputOperator.getOutputSchema();
+        
+        // check if input schema is present
+        if (! inputSchema.containsField(predicate.getInputAttributeName())) {
+            throw new RuntimeException(String.format(
+                    "input attribute %s is not in the input schema %s",
+                    predicate.getInputAttributeName(),
+                    inputSchema.getAttributeNames()));
+        }
+        
+        // check if attribute type is valid
+        AttributeType inputAttributeType = 
+                //Does predicate give the correct input attribute name?
```

Comment:
Remove this question, and yes, Line 87 above checks if the attribute name is valid.